harvard-lts/rabbitmq-email-notifier
# ETD-201:  RabbitMQ version of the Email Notifier

**Create a RabbitMQ version of the Email Notifier**
* * *

**JIRA Ticket**: https://jira.huit.harvard.edu/browse/ETD-201


# What does this Pull Request do?
 RabbitMQ version of the Email Notifier. this is the rabbitmq analog to https://github.com/harvard-lts/mq_email_notifier

# How should this be tested? 
1) copy` .env.example `to `.env `and set to dev settings. set `NEXT_TASK` to a task that exists in a celery pipeline (ie: "`etd-base-template.tasks.send_to_base`")
2) pull down jaeger and run it locally: see https://wiki.harvard.edu/confluence/pages/viewpage.action?spaceKey=LibraryTechServices&title=OpenTelemetry 
3) run image: `docker-compose -f docker-compose-local.yml up -d --build --force-recreate`
4) enter container, and run `pytest`. the integration test should pass.
5) to see a message manually put on the queue, run `python scripts/invoke-task.py` and then go to the rabbitmq console and look at the `dead-letter` queue. its message count should be increased by one after running that script.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? n
- integration tests? y

# Interested parties
Tag (@ mention) interested parties: @awoods
